### PR TITLE
tests: make FilterFactory reuseable

### PR DESCRIPTION
**Pull request by BusyJay**
@ngaut @siddontang @disksing @zhangjinpeng1987 PTAL

### CODE REVIEW

**Comment by ngaut** [comment]
Why we need to  make FilterFactory reusable ?

**Comment by BusyJay** [comment]
Because there are many duplicated code that just wrap a `Filter` and clone it when needed. It can be abstracted as a generic factory and make writing `Filter` more easy and clean.

**Comment by BusyJay** [comment]
PTAL

**Comment by zhangjinpeng1987** [comment]
LGTM

File: tests/raftstore/transport_simulate.rs [hunk 1]
```diff
@@ -293,54 +294,35 @@ impl Filter<RaftMessage> for FilterRegionPacket {
...
-                 direction: self.direction,
-                 msg_type: self.msg_type.clone(),
-                 allow: AtomicUsize::new(self.allow),
-             }]
-    }
-}
-
-struct SnapshotFilter {
+#[derive(Default)]
+pub struct SnapshotFilter {
```
Comment (hhkbp2) [inline tests/raftstore/transport_simulate.rs]: xxxFilter or Filterxxx? Both occur in this file.
IMO, xxxFilter would be a better name for a struct. And may every filter factory should has the suffix 'FilterFactory'.
I think it needs to be uniformed for the naming convention. Maybe we could do it some time later.

**Comment by hhkbp2** [comment]
LGTM

**Pull request merged**